File: easy/53. Maximum Subarray/Solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSoFar = nums[0];
        int maxNow = nums[0];
        for(int i = 1; i< nums.length;i++){
            maxNow = Math.max(nums[i],nums[i]+maxNow);
            maxSoFar = Math.max(maxSoFar,maxNow);
        }
        return maxSoFar;
        
    }
}

File: easy/53. Maximum Subarray/Solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        current = nums[0]
        sofar = nums[0]
        for i in range(1,len(nums)):
            current = max(current+nums[i],nums[i])
            sofar = max(current,sofar)
        return sofar
